### PR TITLE
:bug: Fix touchpad swipe back/forward #4246

### PR DESCRIPTION
**Pull request by dfelinto**
### Related Ticket

Github: #4246
Taiga: https://tree.taiga.io/project/penpot/issue/7454

### Summary

This is intentionally a fix for *most* of the cases. As commented on the code, users may still run into the original issue in some corner cases.

In my tests I need to go from the workspace to the dashboard and back very quickly for this to happen, and it doesn't even happen every time. I think having the ability to still use gestures on the dashboard justifies this quirk.

### Steps to reproduce

1. Go to the dashboard.
2. Open a document.
3. Use the trackpad to pan around.

The browser should *not* navigate back to the dashboard.

_Tested on MacOS + Google Chrome._

### Checklist

- [x] Choose the correct target branch; use `develop` by default.
- [x] Provide a brief summary of the changes introduced.
- [x] Add a detailed explanation of how to reproduce the issue and/or verify the fix, if applicable.
- [ ] ~~Include screenshots or videos, if applicable.~~
- [x] Add or modify existing integration tests in case of bugs or new features, if applicable. ¹
- [x] Check CI passes successfully.
- [x] Update the `CHANGES.md` file, referencing the related GitHub issue, if applicable.

¹ The following tests are failing in the `develop` branch, even without my patch:
```
[default] › playwright/ui/specs/subscriptions-dashboard.spec.js:85:3 › Subscriptions: team members and invitations › Team settings has susbscription name and no manage subscription link when is member 
[default] › playwright/ui/specs/variants.spec.js:114:1 › User duplicates a variant container 
```

### CODE REVIEW

File: frontend/src/app/main/ui.cljs [hunk 1]
```diff
@@ -353,6 +353,15 @@
...
        nil)]))
 
+(defn use-prevent-browser-gesture-navigation!
+  []
+  ;; Prevent the browser from interpreting trackpad horizontal swipe as back/forth
+  ;;
+  ;; In theory We could disable this only for the workspace. However gets too unreliable.
+  ;; It is better to be safe and disable for the dashboard as well.
+  (set! (.. js/document -documentElement -style -overscrollBehaviorX) "none")
+  (set! (.. js/document -body -style -overscrollBehaviorX) "none"))
```
Comment (niwinz) [inline frontend/src/app/main/ui.cljs]: I think we should name it `prevent-browser-gesture-navigation!`  (removing the `use-` prefix because it is not a react hook and right now use- prefix indicates a react hook)

And move it under `app.util.dom` because there we have all the dom manipulation helpers.

**Comment by niwinz** [comment]
Hey, I've leaved a comment, Do you prefer do prefer make the change yourself or also I can do it myself.
In any case, thanks for the contribution :heart:

**Comment by dfelinto** [comment]
I can do it later today.

The name was indeed a left-over from my early attempy of making a hook based on the route, good catch. And app.util.dom it is :) I was indeed struggling a bit finding the right place.

By the way, is it best to do a force rebase on my branch or a "fixup" commit? (will do the latter unless I hear otherwise).

**Comment by dfelinto** [comment]
Hm I think I do need to do the rebase to keep the PR 1:1 with the expected commit.

**Comment by dfelinto** [comment]
I force updated so it passes the github CI.

**Comment by niwinz** [comment]
Yep, force update Is OK :D

**Comment by dfelinto** [comment]
Just to be clear, I addressed the changes you suggested already.

If there are other tweaks let me know. Also I won't mind if you rather tackle things yourself either if it simpler/speeds things.

Either way, thanks for taking the time to review this.
